### ISSUE TITLE
[FR] Configurable CONTROLLER_FAN triggers

### ISSUE THREAD

**Issue by melslenstra** [MarlinFirmware/Marlin#23401]
_This feature request is about the controller fan feature as enabled with `USE_CONTROLLER_FAN`, intended to control fan(s) used to cool the 3D printer motherboard. I've seen a lot of confusion in different issues around which kind of fan control is meant exactly, so I wanted to clarify that first of all._

### Is your feature request related to a problem? Please describe.

I wouldn't want to call fans coming on when not really needed a _problem_, per se. Just a minor inconvenience ;-)

### Are you looking for hardware support?

No.

### Describe the feature you want

The current implementation of `ControllerFan` ([link to relevant lines of code](https://github.com/MarlinFirmware/Marlin/blob/f17f458e919218557bd01c24b1a2eb621ba07241/Marlin/src/feature/controllerfan.cpp#L62)) is hard-coded to look at a few triggers that will cause the controller fan to come on at the configured `CONTROLLERFAN_SPEED_ACTIVE`. The current triggers are:
* Any stepper motor is enabled
* The heated bed is turned on
* The optional designated board temperature sensor is reporting a temperature higher than `CONTROLLER_FAN_MIN_BOARD_TEMP` (note: by not defining `CONTROLLER_FAN_MIN_BOARD_TEMP` this trigger is essentially already configurable)

It would be great if it were possible to **enable or disable the controller fan triggers individually** in `Configuration_adv.h` with settings along the lines of `CONTROLLER_FAN_IGNORE_BED` and `CONTROLLER_FAN_IGNORE_MOTORS`. 

### Additional context

I run a heavily modified Creality Ender 5 Plus with:
* The power supply and heated bed MOSFET daughterboard in the original black bottom housing
* The motherboard (BTT GTR 1.0 + M5 w/ TMC stepper drivers) in a custom laser cut acrylic housing mounted in the side of the frame (already published accompanying TFT43 case [here](https://www.thingiverse.com/thing:4917037), will publish board housing files soon)

My controller fans are two Noctua NF-A4x20 fans in the rear of the motherboard housing blowing air across the motherboard, expansion and Raspberry Pi (OctoPrint). They're connected in parallel to one 12V fan output on the M5 expansion. Although these fans are pretty quiet even at full speed, it's still distinctly audible.

The point of this feature request for me is that the fans kick in at full speed when I preheat the bed, even though the MOSFETs for the bed are nowhere near the fans.

A perhaps even better implementation would be to be able to freely configure an arbitrary number of instances of controller fans with different triggers. But I fully understand that'd be a much bigger deal.

**Comment by thisiskeithb** [comment]
Duplicate of https://github.com/MarlinFirmware/Marlin/issues/11253
